src/Apps/Auction/AuctionApp.tsx

```diff
@@ -0,0 +1,95 @@
+import { Box, Separator, Serif } from "@artsy/palette"
+import { AuctionApp_sale } from "__generated__/AuctionApp_sale.graphql"
+import { AppContainer } from "Apps/Components/AppContainer"
+import { trackPageViewWrapper } from "Apps/Order/Utils/trackPageViewWrapper"
+import { ErrorPage } from "Components/ErrorPage"
+import React, { Component } from "react"
+import { createFragmentContainer, graphql, RelayProp } from "react-relay"
+import {
+  Elements,
+  ReactStripeElements,
+  StripeProvider,
+} from "react-stripe-elements"
+import RegistrationForm from "./Components/RegistrationForm"
+
+declare global {
+  interface Window {
+    Stripe?: (key: string) => stripe.Stripe
+    sd: {
+      STRIPE_PUBLISHABLE_KEY: string
+    }
+  }
+}
+
+interface AuctionAppProps extends ReactStripeElements.InjectedStripeProps {
+  sale: AuctionApp_sale
+  relay?: RelayProp
+}
+
+interface AuctionAppState {
+  stripe: stripe.Stripe
+}
+
+export class AuctionApp extends Component<AuctionAppProps, AuctionAppState> {
+  state: AuctionAppState = { stripe: null }
+  CreditCardInput
+  componentDidMount() {
+    if (window.Stripe) {
+      this.setState({
+        stripe: window.Stripe(window.sd.STRIPE_PUBLISHABLE_KEY),
+      })
+    } else {
+      document.querySelector("#stripe-js").addEventListener("load", () => {
```

Comment:
We should remove this listener on `componentWillUnmount`. Alternatively, if you wanted to use a functional component / hooks, could do something like this: 

```js
useEffect(() => {
  document.querySelector("#stripe-js").addEventListener("load", ...)

  return () => {
    document.querySelector("#stripe-js").removeEventListener("load", ...)
  }
}, [])
```
(No personal pref re classes / function components, just FYI)

Reply:
Looks cool! I'll look into using a functional component / hooks in the next iteration. :+1: